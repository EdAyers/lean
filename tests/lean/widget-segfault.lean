open widget
open lean
open lean.parser
open interactive
open tactic

meta def component.stateless' {π α} (view : π → list (html α)) : component π α :=
component.mk
  α
  unit
  (λ _ _, ())
  (λ _ _ a, (⟨⟩, some a))
  (λ p _, view p )
  (λ _ _, ff)

meta def comp : component tactic_state string :=
component.stateless' $ λ ts,
  [h "h1" [] ["WIDGET"],
    h "img" [attr.val "src" "https://html.com/wp-content/uploads/flamingo.jpg"] []
  ]

@[user_command]
meta def widget_test (_ : parse (tk "#widgety")) : lean.parser unit :=
do  ⟨l,c⟩ ← cur_pos,
    tactic.save_widget ⟨l,c - ("#widgety").length - 1⟩ _, -- underscore here
    pure ()

run_cmd skip

#widgety
