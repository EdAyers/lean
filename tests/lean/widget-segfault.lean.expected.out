widget-segfault.lean:25:55: error: don't know how to synthesize placeholder
context:
_x : parse (tk "#widgety"),
widget_test : parser unit,
_p : pos,
_do_match : pos → parser unit,
l c : ℕ
⊢ component tactic_state string
